app/mailers/school_mailer.rb

```diff
@@ -66,33 +69,27 @@ def nomination_confirmation_email(user:, school:, start_url:, step_by_step_url:)
         subject: "Sign in to manage induction",
         step_by_step: step_by_step_url,
       },
-    )
+    ).tag(:sit_nominated).associate_with(school, sit_profile)
   end
 
-  def school_partnership_notification_email(
-    recipient:,
-    lead_provider_name:,
-    delivery_partner_name:,
-    school_name:,
-    nominate_url:,
-    challenge_url:,
-    challenge_deadline:
-  )
+  # This email is sent to the school which has been reported by the lead provider and which hasn't yet nominated SIT
+  # It also contains request to nominate SIT.
+  def school_partnership_notification_email(recipient:, partnership:, challenge_url:, nominate_url:)
     template_mail(
       SCHOOL_PARTNERSHIP_NOTIFICATION_EMAIL_TEMPLATE,
       to: recipient,
       rails_mailer: mailer_name,
       rails_mail_template: action_name,
       personalisation: {
-        lead_provider_name: lead_provider_name,
-        delivery_partner_name: delivery_partner_name,
-        school_name: school_name,
+        lead_provider_name: partnership.lead_provider.name,
+        delivery_partner_name: partnership.delivery_partner.name,
+        school_name: partnership.school.name,
         nominate_url: nominate_url,
         challenge_url: challenge_url,
-        challenge_deadline: challenge_deadline,
+        challenge_deadline: partnership.challenge_deadline.strftime("%d/%m/%Y"),
         subject: "FAO: NQT coordinator. Training provider confirmed.",
       },
-    )
+    ).tag(:partnership_created, :request_to_nominate_sit).associate_with(partnership, partnership.school)
```

Comment:
Out of curiosity, can this `associate_with` be called where the mailer is called? As in `SchoolMailer.school_partnership_notification_email(...).associate_with(...).deliver_later`? Or is the mailer magic in a way that prevents that?

Reply:
Ha! Just commented the same thing :D

So yes, it is theratically possible, but the way mailer deliver_later works is that it stores a ActionMailer::MailDeliveryJob (serialized and wrapped two time), which just keeps trakc of the mailer class, email name and arguments to the mailer. When executed in the backend it just invokes `MailerClass.email_name(**arguments).deliver_now` - so to make tagging and associate_with working outside of the mailer I'd need to extend this job class - and extending Mail::Message already brought quite a lot of pushback...